api/src/main/java/io/minio/MinioClient.java

```diff
@@ -762,6 +762,19 @@ public boolean verify(String hostname, SSLSession session) {
   
 
   /**
+   * Set customized HTTP client.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code minioClient.setHttpClient(customHttpClient); }</pre>
+   *
+   * @param customHttpClient Customized HTTP client object.
+   */
+  public void setHttpClient(OkHttpClient customHttpClient) {
+    this.httpClient = customHttpClient;
```

Comment:
Passing null will throw `NullPointerException` at runtime.  This method is for specific use case I think its OK.